pygmt/datasets/load_remote_dataset.py

```diff
@@ -269,43 +266,44 @@ def _load_remote_dataset(
 
     Note
     ----
-    The returned :class:`xarray.DataArray` doesn't support slice operation for
-    tiled grids.
+    The returned :class:`xarray.DataArray` doesn't support slice operation for tiled
+    grids.
     """
     dataset = datasets[dataset_name]
 
     # check resolution
-    if resolution not in dataset.resolutions:
+    for res in dataset.resolutions:
+        if res.code == resolution:
+            valid_registrations = res.registrations
+            is_tiled = res.tiled
+            break
+    else:
         raise GMTInvalidInput(f"Invalid resolution '{resolution}'.")
```

Comment:
This for-loop seems a lot more inefficient compared to the previous `if resolution not in dataset.resolutions` dictionary-based lookup. I almost think we should go with the intermediate `{"01d": Resolution(), ...}` dictionary you mentioned at https://github.com/GenericMappingTools/pygmt/pull/2917#issue-2056550622, or come up with a better data structure than a list of `Resolution` NamedTuples. I almost feel like that `datasets` variable could be a nested JSON or a multi-level pandas.DataFrame object.

Reply:
Alternatively, we could just keep the `datasets` variable as is, but bring in all the type hint stuff. The remote datasets aren't really updated that often, though I know you're working on those Moon/Venus/Mars PRs.

Reply:
> This for-loop seems a lot more inefficient compared to the previous `if resolution not in dataset.resolutions` dictionary-based lookup. 

Yes, it's almost 10-times slower.

> I almost think we should go with the intermediate `{"01d": Resolution(), ...}` dictionary you mentioned at [#2917 (comment)](https://github.com/GenericMappingTools/pygmt/pull/2917#issue-2056550622), or come up with a better data structure than a list of `Resolution` NamedTuples. I almost feel like that `datasets` variable could be a nested JSON or a multi-level pandas.DataFrame object (but maybe that's overkill).

What about a dictionary:
```
{
   "01d": Resolution("01d"),
}
```
It's more clear than:
```
{
    "01d": Resolution(),
}
```

Reply:
> What about a dictionary:\n\n{\n   \"01d\": Resolution(\"01d\"),\n}

Yes, that could work.